packages/@aws-cdk/aws-stepfunctions/lib/states/map.ts

```diff
@@ -150,10 +150,30 @@ export class Map extends State implements INextable {
      * Validate this state
      */
     protected validate(): string[] {
-        if (!!this.iterator) {
-            return ['Map state must have a non-empty iterator'];
+        const validateMaxConcurrency = () => {
```

Comment:
Not sure why this needs to be an inner function? I'd rather you extract the logic to test that a number is a positive integer to a function `isPositiveInteger()` and call that inline in this function.

Reply:
Ok i will move the function outside Map and export it for the test, I will add tests for the function (the samecases  I added) but I would like to leave the tests on synth to take care about regressions